hamilton/function_modifiers/recursive.py

```diff
@@ -626,3 +629,314 @@ def prune_nodes(nodes: List[node.Node], select: Optional[List[str]] = None) -> L
                 stack.append(dep_node)
             seen_nodes.add(dep)
     return output
+
+
+class with_columns(base.NodeInjector, abc.ABC):
```

Comment:
This is abstract -- where are the functions that are left as abstract? E.G. what does a subclass have to implement?